lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -440,6 +440,213 @@ class AutofillInfo {
   }
 }
 
+/// Replaces a range of text in the original string with the text given in the
+/// replacement string.
+String _replace(String originalText, String replacementText, ui.TextRange replacedRange) {
+  assert(replacedRange.isValid);
+  assert(replacedRange.start <= originalText.length && replacedRange.end <= originalText.length);
+
+  final ui.TextRange normalizedRange = ui.TextRange(start: math.min(replacedRange.start, replacedRange.end), end: math.max(replacedRange.start, replacedRange.end));
+
+  final String newText = normalizedRange.textBefore(originalText) + replacementText + normalizedRange.textAfter(originalText);
```

Comment:
Nit: Maybe just return this directly instead of assigning to a variable.

Reply:
sgtm